src/reanimated2/layoutReanimation/web/animationsManager.ts

```diff
@@ -141,10 +125,14 @@ export function startWebLayoutAnimation<
     initialAnimationName as AnimationNames
   );
 
-  if (checkReduceMotionFail(animationConfig, hasEnteringAnimation, element)) {
+  if (checkReduceMotionFail(animationConfig)) {
     return;
   }
 
+  if (shouldMakeVisible) {
+    makeElementVisible(element);
+  }
+
```

Comment:
Correct me if I'm wrong, but now if user has enabled reduced motion (or we encounter undefined animation) element will stay `hidden`, because that part of code won't be reached.